2-WebApp-graph-user/2-5-HybridFlow/README.md

```diff
@@ -0,0 +1,771 @@
+     * You'll need this key later in your code's configuration files. This key value will not be displayed again, and is not retrievable by any other means, so make sure to note it from the Azure portal before navigating to any other screen or blade.
+  3. In the app's registration screen, select the **API permissions** blade in the left to open the page where we add access to the APIs that your application needs.
+     * Select the **Add a permission** button and then,
+     * Ensure that the **Microsoft APIs** tab is selected.
+     * In the *Commonly used Microsoft APIs* section, select **Microsoft Graph**
+     * In the **Delegated permissions** section, select the **User.Read** in the list followed by **Contacts.Read**. Use the search box if necessary.
+     * Select the **Add permissions** button at the bottom.
+
+##### Configure the application to use your registration
+
+  Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+   > In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+  1. Open the `appsettings.json` file.
+  1. Find the key `Domain` and replace the existing value with your Azure AD tenant name.
+  1. Find the key `TenantId` and replace the existing value with your Azure AD tenant ID.
+  1. Find the key `ClientId` and replace the existing value with the application ID (clientId) of `HybridFlow-aspnetcore` app copied from the Azure portal.
+  1. Find the key `ClientSecret` and replace the existing value with the **client secret** of `HybridFlow-aspnetcore` app copied from the Azure portal.
+
+#### (Optional) Create a self-signed certificate
+
+  To complete this step, you will use the `New-SelfSignedCertificate` Powershell command. You can find more information about the New-SelfSignedCertificate command [here](https://docs.microsoft.com/powershell/module/pkiclient/new-selfsignedcertificate).
+  
+  1. Open PowerShell and run `New-SelfSignedCertificate` with the following parameters to create a self-signed certificate in the user certificate store on your computer:
+  
+      ```PowerShell
+      $cert=New-SelfSignedCertificate -Subject "CN=HybridFlowCert" -CertStoreLocation "Cert:\CurrentUser\My"  -KeyExportPolicy Exportable -KeySpec Signature
+      ```
+  
+  1. Export this certificate using the "Manage User Certificate" MMC snap-in accessible from the Windows Control Panel. You can also add other options to generate the certificate in a different
+  store such as the Computer or service store (See [How to: View Certificates with the MMC Snap-in](https://docs.microsoft.com/dotnet/framework/wcf/feature-details/how-to-view-certificates-with-the-mmc-snap-in)).
+  
+  Alternatively you can use an existing certificate if you have one (just be sure to record its name for the next steps)
+  
+  ### Add the certificate for the application in Azure AD
```

Comment:
>### [](http://example.com/codeflow?start=2&length=3)

since this and other below in under 
#### (Optional) Create a self-signed certificate

so they cannot have less '#' than the title. they should either have #### or more in their titles. 

[Potential code gen fix?]